include/seqan/bam_io/bam_index_bai.h

```diff
@@ -154,6 +154,160 @@ class BamIndex<Bai>
 // Functions
 // ============================================================================
 
+// ----------------------------------------------------------------------------
+// Function viewRecords()
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn BamFileIn#viewRecords
+ * @brief Extract all reads overlapping a region from a BamFileIn using an index.
+ *
+ * You provide a region <tt>[regionStart, regionEnd]</tt> on the reference <tt>refID</tt>
+ * and the functions extracts all reads that overlap this region with at least
+ * one base. The overlapping records are **appended** in the resultContainer, which
+ * must be a container over BamAlignmentRecord (Value type).
+ *
+ * @signature bool viewRecords(bamFileIn, bamIndex, refID, regionStart, regionEnd, index);
+ *
+ * @param[in,out] resultContainer Container of @link BamAlignmentRecord @endlink 's to store the alignments.
+ * @param[in,out] bamFileIn       The @link BamFileIn @endlink to jump with.
+ * @param[in]     bamIndex        The @link BamIndex @endlink to use for the jumping.
+ * @param[in]     refID           The reference id to jump to (<tt>int32_t</tt>).
+ * @param[in]     regionStart     The begin of the region to jump to (<tt>int32_t</tt>).
+ * @param[in]     regionEnd       The end of the region to jump to (<tt>int32_t</tt>).
+ *
+ * @return bool true if seeking was successful, false if an error occurred or no
```

Comment:
Hm yeah lets discuss 😄 
I agree that returning a bool might not be the perfect solution, but 
1) isn't the SeqAn2 "style" an in-out-parameter instead of returning it? Even `readRecord()` gets the record passed as in-out instead of returning it.
 2) Wouldn't I force the user to use a specific container (e.g. `String<BamAlignmentRecord>`) that he eventually wants to change (e.g. vector, array or list) thereby copying?
3) I deliberately documented to **append** the reads to the container such that the user can call  `viewRecords(container, ..region1..); viewRecords(container, ..region2..)` and has all records of both regions in the container. I thought this might be useful. And it might be more intuitive than appending a return value, no?

Just my thoughts on this before we change it. 
(PS: If we do return a container I'd prefer a std::vector..)